htdocs/core/lib/company.lib.php

```diff
@@ -406,11 +410,22 @@ function getState($id,$withcode='',$dbtouse=0)
         $obj = $dbtouse->fetch_object($resql);
         if ($obj)
         {
-            $label=$obj->label;
-            if ($withcode == '1') return $label=$obj->code?"$obj->code":"$obj->code - $label";
-            else if ($withcode == '2') return $label=$obj->code;
-            else if ($withcode == 'all') return array('id'=>$obj->rowid,'code'=>$obj->code,'label'=>$label);
```

Comment:
Also, $withcode == 'all' is not used anywhere on the rest of the code. Should I leave it alone? 
Showing rowid in a select field seems kind of pointless to me

Reply:
I found it. 'all' as not been implemented yet in getState(), the behavior should be like getCountry():
$tmparray=getCountry(GETPOST('country_id','int'),'all',$db,$langs,0);
	if (! empty($tmparray['id']))
	{
		$mysoc->country_id   =$tmparray['id'];
		$mysoc->country_code =$tmparray['code'];
		$mysoc->country_label=$tmparray['label'];

		$s=$mysoc->country_id.':'.$mysoc->country_code.':'.$mysoc->country_label;
		dolibarr_set_const($db, "MAIN_INFO_SOCIETE_COUNTRY", $s,'chaine',0,'',$conf->entity);
	}

Right?